test/testscript.go

```diff
@@ -141,6 +184,39 @@ func (e *ProjectTest) ExecuteMakeTarget(target string) *ProjectTest {
 	return e
 }
 
+func (e *ProjectTest) NPMInstallDeps() *ProjectTest {
+	setupFunc := func(env *testscript.Env) error {
+		// Configure both npm and npx to only create and write files to the
+		// current working directory instead of the default $HOME directory
+		env.Setenv("NPM_CONFIG_CACHE", env.WorkDir+"/.npm")
+		env.Setenv("NPM_CONFIG_PREFIX", env.WorkDir+"/.npm-global")
+		env.Setenv("XDG_CACHE_HOME", env.WorkDir+"/.npx")
```

Comment:
Also are you sure that these are correct?  This seems to be setting XDG confgs to npm/npx specifics folders.  For more on the XDG specification see https://specifications.freedesktop.org/basedir-spec/latest/

Reply:
Discussed over vc. Still setting `XDG` but it is more explicit now that it happens up in the test function.